main.py

```diff
@@ -98,27 +104,33 @@ async def on_message(message: Message):
         await command_stop()
 
 
-# Take a filename as string and return it formatted nicely
-def format_filename(filename: str):
+# Format a song as text nicely using artist/title tags if available
+# local_filepath is the actual path on disc
+# filename is the filename on Discord
+# artist_fallback is the fallback author value (no fallback if not passed)
+def song_format(local_filepath: str, filename: str, artist_fallback: str = ""):
```

Comment:
I would probably default `artist_fallback` to `None` rather than an empty string. Especially as we check for `None` below.

Of course `None` isn't of the `str` type. In this case though, you can make use of the [Optional](https://docs.python.org/3/library/typing.html#typing.Optional) type. Which can be either `<some type>` or `None`. For instance: `artist_fallback: Optional[str] = None`.